scylla/src/lib.rs

```diff
@@ -131,8 +131,89 @@ pub mod frame {
 }
 
 /// Serializing bound values of a query to be sent to the DB.
+// I don't know any other way to tell rustfmt to not reorder
+// imports here.
+#[rustfmt::skip] 
```

Comment:
I was only able to come up with something like this (we can't achieve the type per line formatting this way, though):
```rust
/// Serializing bound values of a query to be sent to the DB.
pub mod serialize {
    pub use scylla_cql::types::serialize::SerializationError;
    pub mod batch {
        // Main types
        pub use scylla_cql::types::serialize::batch::{
            BatchValues, BatchValuesFromIterator, BatchValuesIterator,
            BatchValuesIteratorFromIterator, TupleValuesIter,
        };

        // Legacy migration types - to be removed when removing legacy framework
        pub use scylla_cql::types::serialize::batch::{
            LegacyBatchValuesAdapter, LegacyBatchValuesIteratorAdapter,
        };
    }

    pub mod row {
        // Main types
        pub use scylla_cql::types::serialize::row::{RowSerializationContext, SerializeRow};

        // Errors
        pub use scylla_cql::types::serialize::row::{
            BuiltinTypeCheckError, BuiltinTypeCheckErrorKind,
        };

        // Legacy migration types - to be removed when removing legacy framework
        pub use scylla_cql::types::serialize::row::{
            // Legacy migration types - to be removed when removing legacy framework
            serialize_legacy_row,
            BuiltinSerializationError,
            BuiltinSerializationErrorKind,
        };

        // Not part of the old framework, but something that we should
        // still aim to remove.
        pub use scylla_cql::types::serialize::row::{
            SerializedValues, SerializedValuesIterator, ValueListAdapter,
            ValueListToSerializeRowAdapterError,
        };
    }

    pub mod value {
        // Main types
        pub use scylla_cql::types::serialize::value::SerializeValue;

        // Errors
        pub use scylla_cql::types::serialize::value::{
            BuiltinSerializationError, BuiltinSerializationErrorKind, BuiltinTypeCheckError,
            BuiltinTypeCheckErrorKind, MapSerializationErrorKind, MapTypeCheckErrorKind,
            SetOrListSerializationErrorKind, SetOrListTypeCheckErrorKind,
            TupleSerializationErrorKind, TupleTypeCheckErrorKind, UdtSerializationErrorKind,
            UdtTypeCheckErrorKind,
        };

        // Legacy migration types - to be removed when removing legacy framework
        pub use scylla_cql::types::serialize::value::{
            serialize_legacy_value, ValueAdapter, ValueToSerializeValueAdapterError,
        };
    }

    pub mod writers {
        pub use scylla_cql::types::serialize::writers::{
            CellOverflowError, CellValueBuilder, CellWriter, RowWriter, WrittenCellProof,
        };
    }
}
```

Reply:
I don't love the need to repeat `pub use scylla_cql::types::serialize::something` so many times, but it is probably the best solution for now. 

I'm starting to get really frustrated with rustfmt lately...

Reply:
I applied the sugestion.
Thenks to it I found that earlier reorders by rustfmt caused some types to land in the wrong category - I fixed that too.